mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -377,6 +377,26 @@ def NVVM_Barrier0Op : NVVM_Op<"barrier0"> {
   let assemblyFormat = "attr-dict";
 }
 
+def NVVM_BarrierOp : NVVM_Op<"barrier"> {
+  let arguments = (ins     
+    DefaultValuedAttr<ConfinedAttr<I32Attr, [IntMinValue<0>, IntMaxValue<15>]>, "0">:$barrierResource,
```

Comment:
I agree that we mostly use this as an immediate argument.
But PTX spec also says this can be a value in a register also.
https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-bar-barrier

So, should we make this an operand instead of an attribute ?

Reply:
Good catch, yes in this case it is better to make this as an operand.